chia/consensus/blockchain_interface.py

```diff
@@ -1,103 +1,48 @@
-from chia.types.blockchain_format.vdf import VDFInfo
 from chia.types.header_block import HeaderBlock
 from chia.types.weight_proof import SubEpochChallengeSegment
 from chia.util.ints import uint32
 
 
-class BlockchainInterface:
-    def get_peak(self) -> Optional[BlockRecord]:
-        pass
-
-    def get_peak_height(self) -> Optional[uint32]:
-        pass
-
-    def block_record(self, header_hash: bytes32) -> BlockRecord:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    def height_to_block_record(self, height: uint32) -> BlockRecord:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    def get_ses_heights(self) -> List[uint32]:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    def get_ses(self, height: uint32) -> SubEpochSummary:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    def height_to_hash(self, height: uint32) -> Optional[bytes32]:
-        pass
-
-    def contains_block(self, header_hash: bytes32) -> bool:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    async def contains_block_from_db(self, header_hash: bytes32) -> bool:
-        return  # type: ignore[return-value]
-
-    def remove_block_record(self, header_hash: bytes32) -> None:
-        pass
-
-    def add_block_record(self, block_record: BlockRecord) -> None:
-        pass
-
-    def contains_height(self, height: uint32) -> bool:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    async def warmup(self, fork_point: uint32) -> None:
-        pass
-
-    async def get_block_record_from_db(self, header_hash: bytes32) -> Optional[BlockRecord]:
-        pass
-
-    async def get_block_records_in_range(self, start: int, stop: int) -> Dict[bytes32, BlockRecord]:
-        # ignoring hinting error until we handle our interfaces more formally
-        return  # type: ignore[return-value]
-
-    async def prev_block_hash(self, header_hashes: List[bytes32]) -> List[bytes32]:
-        return  # type: ignore[return-value]
+class BlockRecordsProtocol(Protocol):
+    def try_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]: ...
+    def block_record(self, header_hash: bytes32) -> BlockRecord: ...
+    def contains_height(self, height: uint32) -> bool: ...
+    def contains_block(self, header_hash: bytes32) -> bool: ...
+    def height_to_hash(self, height: uint32) -> Optional[bytes32]: ...
+    def height_to_block_record(self, height: uint32) -> BlockRecord: ...
+
+
+class BlockchainInterface(BlockRecordsProtocol, Protocol):
```

Comment:
```suggestion
class BlockchainProtocol(BlockRecordsProtocol, Protocol):
```

Reply:
this would be a bigger change.
```
$ git grep BlockchainInterface | wc -l
      24
```

Reply:
I suppose I could add another commit on top

Reply:
however you want to handle it including handling it separate from this pr